java/maven.indexer/src/org/netbeans/modules/maven/indexer/NexusRepositoryIndexerImpl.java

```diff
@@ -958,47 +899,39 @@ private void iterate(List<RepositoryInfo> repos, final RepoAction action, final
                         LOGGER.log(Level.INFO, "could not process " + repo.getId(), x);
                     }
                     return null;
-                }
-            });
+                });
+            }
         }
     }
-    }
 
     private ResultImplementation<String> filterGroupIds(final String prefix, final List<RepositoryInfo> repos) {
-        Redo<String> redo = new Redo<String>() {
-            @Override
-            public void run(ResultImpl<String> result) {
-                filterGroupIds(prefix, result, result.getSkipped(), false);
-            }
-        };
-        ResultImpl<String> result = new ResultImpl<String>(redo);
+        ResultImpl<String> result = new ResultImpl<>(
+            (ResultImpl<String> res) -> filterGroupIds(prefix, res, res.getSkipped(), false)
+        );
         return filterGroupIds(prefix, result, repos, true);
     }
     
     private ResultImplementation<String> filterGroupIds(final String prefix, final ResultImpl<String> result, 
                                                             final List<RepositoryInfo> repos, final boolean skipUnIndexed) {
-        final Set<String> groups = new TreeSet<String>(result.getResults());
-        final List<RepositoryInfo> slowCheck = new ArrayList<RepositoryInfo>();
+        final Set<String> groups = new TreeSet<>(result.getResults());
+//        final List<RepositoryInfo> slowCheck = new ArrayList<>();
```

Comment:
Should slowCheck be removed? Comment below says, that is is used rarely, now it is commented out - maybe it should be removed?

Reply:
removed it.